contracts/modules/TransferManager/VolumeRestrictionTM.sol

```diff
@@ -0,0 +1,934 @@
+            _startTime,
+            1,
+            _endTime,
+            RestrictionType(_restrictionType)
+        );
+        emit ModifyDailyGlobalRestriction(
+            _allowedTokens,
+            _allowedPercentageOfTokens,
+            _startTime,
+            1,
+            _endTime,
+            _restrictionType
+        );
+    }
+
+    /**
+     * @notice Internal function have a logic to validate the txn amount with global restriction 
+     */
+    function _globalRestrictionCheck(address _from, uint256 _amount, bool _isTransfer) internal returns (Result) {
+        uint256[] memory timestamps = globalBucketDetails.timestamps;
+        uint256 fromTimestamp;
+        uint256 sumOfLastPeriod = 0;
+        if (timestamps.length == 0) {
+            // It will execute when the txn is performed first time after the addition of global restriction
+            fromTimestamp = globalRestriction.startTime;
+        } else {
+            // picking up the preivous timestamp
+            fromTimestamp = timestamps[timestamps.length -1];
+        }
+        // Calculating the difference of days
+        uint256 diffDays = BokkyPooBahsDateTimeLibrary.diffDays(fromTimestamp, now); 
+        uint256[] memory passedTimestamps = new uint256[](diffDays);
+        // Check with the bucket and parse all the new timestamps to calculate the sumOfLastPeriod
+        // re-using the local variables to avoid the stack too deep error.
+        (sumOfLastPeriod, fromTimestamp, passedTimestamps) = _bucketCheck(
+            fromTimestamp,
+            diffDays,
+            _from,
+            globalBucketDetails,
+            true,
+            globalRestriction
+        );
+        // Total amout that is transacted uptill now for `fromTimestamp` day
+        uint256 txSumOfDay = globalBucket[fromTimestamp];
+        // Global transaction check
+        if (_globalTxCheck(sumOfLastPeriod, txSumOfDay, _amount)) {
+            // allow modification in storage when `_isTransfer` equals true
+            if(_isTransfer) {
+                // update the global storage
+                _updateGlobalStorage(
+                    passedTimestamps,
+                    fromTimestamp,
+                    _amount,
+                    diffDays,
+                    sumOfLastPeriod
+                );
+            }
+            return Result.NA;
+        }
+        return Result.INVALID;
+    }
+
+    /**
+     * @notice Internal function to update the state variables related to global restriction
+     */
+    function _updateGlobalStorage(
+        uint256[] passedTimestamps,
+        uint256 _fromTime,
+        uint256 _amount,
+        uint256 _diffDays,
+        uint256 _sumOfLastPeriod
+    )
+        internal
+    {
+        if (_diffDays != 0) {
+            for (uint256 i = 0; i < passedTimestamps.length; i++) {
+                // Add the timestamp that is already passed
+                globalBucketDetails.timestamps.push(passedTimestamps[i]);
```

Comment:
This also puts the same gas limitation as individual one but it is less likely to occure in real life.